File: db4onqopt/core/src/com/db4o/nativequery/expr/cmp/ArrayAccessValue.java
/* Copyright (C) 2004 - 2006  db4objects Inc.  http://www.db4o.com */

package com.db4o.nativequery.expr.cmp;

public class ArrayAccessValue extends ComparisonOperandDescendant {
	private ComparisonOperand _index;
	
	public ArrayAccessValue(ComparisonOperandAnchor parent,ComparisonOperand index) {
		super(parent);
		_index = index;
	}

	public void accept(ComparisonOperandVisitor visitor) {
		visitor.visit(this);
	}
	
	public ComparisonOperand index() {
		return _index;
	}
	
	public boolean equals(Object obj) {
		if(!super.equals(obj)) {
			return false;
		}
		ArrayAccessValue casted=(ArrayAccessValue)obj;
		return _index.equals(casted._index);
	}
	
	public int hashCode() {
		return super.hashCode()*29+_index.hashCode();
	}
	
	public String toString() {
		return super.toString()+"["+_index+"]";
	}
}

File: db4on/core/src/core/com/db4o/nativequery/expr/cmp/ArrayAccessValue.cs
namespace com.db4o.nativequery.expr.cmp
{
	public class ArrayAccessValue : com.db4o.nativequery.expr.cmp.ComparisonOperandDescendant
	{
		private com.db4o.nativequery.expr.cmp.ComparisonOperand _index;

		public ArrayAccessValue(com.db4o.nativequery.expr.cmp.ComparisonOperandAnchor parent
			, com.db4o.nativequery.expr.cmp.ComparisonOperand index) : base(parent)
		{
			_index = index;
		}

		public override void Accept(com.db4o.nativequery.expr.cmp.ComparisonOperandVisitor
			 visitor)
		{
			visitor.Visit(this);
		}

		public virtual com.db4o.nativequery.expr.cmp.ComparisonOperand Index()
		{
			return _index;
		}

		public override bool Equals(object obj)
		{
			if (!base.Equals(obj))
			{
				return false;
			}
			com.db4o.nativequery.expr.cmp.ArrayAccessValue casted = (com.db4o.nativequery.expr.cmp.ArrayAccessValue
				)obj;
			return _index.Equals(casted._index);
		}

		public override int GetHashCode()
		{
			return base.GetHashCode() * 29 + _index.GetHashCode();
		}

		public override string ToString()
		{
			return base.ToString() + "[" + _index + "]";
		}
	}
}
